src/functions_math.jl

```diff
@@ -67,3 +67,25 @@ function Base.inv(nda::NamedDimsArray{L, T, 2}) where {L,T}
     names = reverse(L)
     return NamedDimsArray{names}(data)
 end
+
+# Statistics
+for fun in (:cor, :cov)
+    @eval function Statistics.$fun(a::NamedDimsArray{L, T, 2}; dims=1, kwargs...) where {L, T}
+        numerical_dims = dim(a, dims)
+        data = Statistics.$fun(parent(a); dims=numerical_dims, kwargs...)
+        names = symmetric_names(L, numerical_dims)
+        return NamedDimsArray{names}(data)
+    end
+end
+
+function symmetric_names(L::Tuple{Symbol,Symbol}, dims::Integer)
+    # 0 Allocations. See `@btime (()-> symmetric_names((:foo, :bar), 1))()`
+    names = if dims == 1
+        (L[2], L[2])
+    elseif dims == 2
+        (L[1], L[1])
+    else
+        (:_, :_)
+    end
```

Comment:
Should we generalize this? 

```
function symmetric_names(L::NTuple{N,Symbol} dim::Integer)
    name = dim <= N ? L[dim] : :_
    names = ntuple(i->name, N)
    return compile_time_return_hack(names)
end
```

Do check performance if so

Reply:
Maybe... Will this ever make sense for the `N != 2` case?

Reply:
if we decide to do, then your suggestion also doesn't allocate :)

```
function symmetric_names(L::NTuple{N, Symbol}, dim::Integer) where N
    name = dim <= N ? L[dim] : :_
    names = ntuple(i -> name, N)
    return compile_time_return_hack(names)
end
```

Reply:
I think my solution is a bit cleaner, but your call

Reply:
actually on second look, the suggestion gives the wrong result

`symmetric_names((:a, :b), 1)` should be `(:b, :b)`, but the suggestion gives `(:a, :a)`

Reply:
Good reason to do it your way